### PR TITLE
Template notebook for looking at AUCell output

### LINKED ISSUE #565: Create a template report for evaluating tumor cell annotations
### If you are filing this issue based on a specific GitHub Discussion, please link to the relevant Discussion.

#292 
Step 2 to address #563 

### Describe the goals of the changes to the analysis module.

Once we have annotated tumor cells in a sample, we want to have some sort of output that we can use to evaluate those annotations. We should make a template report that will be run after the script created in #564 that will output some basic plots to validate our findings. 

### What will your pull request contain?

A template report that ultimately will be part of a workflow to run #564 and generate this report for each sample in SCPCP000015. 

I think we will want to include: 

- A UMAP showing cells as tumor or normal 
- If using `AUCell` as I anticipate we will, a density plot showing the distribution of the AUC in the query sample and in the ref sample (SCPCL000822). 
- Density plots showing marker gene expression and gene set score distributions in tumor and other cells. 
- Heatmaps showing marker gene expression and gene set scores in individual cells. Here we expect to see tumor and normal cells clustering together with distinct expression/ score profiles. 

### Will you require additional software beyond what is already in the analysis module?

No

### Will you require different computational resources beyond what the analysis module already uses?

No

### If known, when do you expect to file the pull request?

_No response_

### PR DESCRIPTION
**Pull request by allyhawkins**
#### Please link to the GitHub issue that this pull request addresses.
Closes #565 

#### What is the goal of this pull request?

This PR adds in a template notebook that will be rendered after running the `01-run-aucell.R` script added in #569. The goal of this template notebook is to have something to look at for each sample in the project and evaluate the tumor cell classification by `AUCell`. 

#### Briefly describe the general approach you took to achieve this goal.

This notebook will take in the following inputs: 

- The SCE object of interest. 
- The results tsv from running `01-run-aucell.R`. 
- The gene set scores calculated using `calculate-gene-set-scores.R`
- The marker genes tsv file 
- The AUC threshold used to classify tumor cells 
- The results tsv from running `01-run-aucell.R` on the reference sample (this will be SCPCL000822). 

I kept things pretty simple here because I just want to see if the tumor cell classifications by `AUCell` make sense. I Included the following plots: 

- A UMAP showing the tumor cell annotations using AUCell. 
- A density plot showing the AUC distribution for the library of interest as compared to the reference. I think including this will be helpful to indicate if using the AUC threshold from the reference is going to be appropriate for each of the samples. 
- Validation of marker gene expression and gene set scores in tumor cells using both density and heatmaps. Here I would expect to see separation of the tumor and normal cells. 

Are there more plots that we should include? I tried to pick those that would be the most informative for deciding if we trust the annotations or not. 

Here's a copy of a rendered version of the report: 

[01-auc-results.html.zip](https://github.com/user-attachments/files/16147199/01-auc-results.html.zip)

#### If known, do you anticipate filing additional pull requests to complete this analysis module?

Yes

### Provide directions for reviewers

_In this section, tell reviewers what kind of feedback you are looking for._
_This information will help guide their review._

#### What are the software and computational requirements needed to be able to run the code in this PR?

Just ran this locally in the Rproj. 

#### Is there anything that you want to discuss further?

Another thing I did while I was here is start a little bit of reorganization. I created separate folders within the `template_notebooks` folder for each of the workflows so we don't have multiple `01` notebooks in one folder. 

I also renamed the `annotate-tumor-cells.sh` to `cnv-annotation.sh`. I know it doesn't _only_ run CNV methods, but that is the main purpose of that workflow. I did this in preparation to add the AUCell workflow as I don't want that one to be mistaken for the "main" workflow. Before actually creating the AUCell workflow, I'm going to work on #572 to finish the documentation updates and any other reorg that should happen. 

### Author checklists

_Check all those that apply._
_Note that you may find it easier to check off these items after the pull request is actually filed._

#### Analysis module and review

- [ ] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [ ] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [ ] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [x] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [ ] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

### CODE REVIEW

File: analyses/cell-type-ewings/template_notebooks/auc-workflow/01-auc-results.Rmd [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+### Heatmap
+
+Here every row is a marker gene and every column is a cell with colors corresponding to the expression of a specified marker gene.  
+The annotation bar indicates which cells are tumor or normal cells as classified by `AUCell`. 
+We expect to see higher expression of individual marker genes in tumor cells compared to normal cells. 
+
+```{r}
+# create matrix with marker genes as rows and barcodes as columns
+marker_gene_heatmap <- markers_df |>
```
Comment (jashapiro) [inline analyses/cell-type-ewings/template_notebooks/auc-workflow/01-auc-results.Rmd:168]: Just for a bit of extra clarity about what is in here, perhaps.
```suggestion
marker_gene_matrix <- markers_df |> 
```

File: analyses/cell-type-ewings/template_notebooks/auc-workflow/01-auc-results.Rmd [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+### Heatmap 
+
+Here every column is a cell and every row is a gene set with colors corresponding to the gene set score. 
+The annotation bar indicates which cells are tumor or normal cells as classified by `AUCell`. 
+We expect to see higher gene set scores in tumor cells compared to normal cells. 
+
+```{r}
+# make a matrix of gene set by barcode 
+geneset_heatmap <- geneset_plot_df |>
```
Comment (jashapiro) [inline analyses/cell-type-ewings/template_notebooks/auc-workflow/01-auc-results.Rmd:238]: Same comment as above
```suggestion
geneset_matrix <- geneset_plot_df |> 
```

**Review by jashapiro — COMMENTED** [review]
This looks good. I had a few minor suggestions.

The one that probably is the most work being changing the tumor/normal order in the legend, but I really don't care much about that either way.

The only other thing is that you probably want to finish off #581 first? There seems to be some overlap between these. But I'll leave it to you about how best to handle that.

**Comment by allyhawkins** [comment]
> In all of these plots, I kind of want to have "Tumor" be first, which means converting to factors, but I don't know that it is worth it. For now it probably doesn't matter...

This was a pretty easy fix, so I went ahead and did it! I also incorporated your other edits so this should be ready for another look.

**Review by jashapiro — APPROVED** [review]
> This was a pretty easy fix, so I went ahead and did it!

Glad it was easy. I guess I thought there might have to be a couple places to update, but I guess just the one does the job! (I'm assuming the heatmaps follow this order too, which is what I was worried would be more of a pain.)

**Pull request merged**